### ISSUE TITLE
Catalog Displays Ghost Nodes

### ISSUE THREAD

**Issue by orarnon** [hashicorp/consul#6474]
#### Overview of the Issue

We're having the same issue again.
We see two instances using the catalog API but only one of them is valid and it is the one that is registered as a member.
The "sonic-is-ds-online" node is the node which is registered as a member while the "ua-producer" node points to the same IP (according to catalog). The 2nd node was de-registered from Consul on Jul 27th.
When using Consul members, grepping for the IP in question:
```
root@consul1:~# consul members | grep "10.200.7.74"
is-ds-online-i-075ec8f8a5438186d.e.sonic-us.us-east-1          10.200.7.74:8301     alive   client  1.4.4  2         sonic-us_us-east-1  <default>
```

We also had another node which appeared in catalog but not in members. Which have de-registered it using
```
curl -i -X PUT -d '{"Node":"<node1>"}' 'http://localhost:8500/v1/catalog/deregister'
```

Might be related to #https://github.com/hashicorp/consul/issues/5220

### Consul info for both Client and Server

<details>
  <summary>Server info</summary>

```
agent:
	check_monitors = 0
	check_ttls = 5
	checks = 5
	services = 6
build:
	prerelease =
	revision = ea5210a3
	version = 1.4.4
consul:
	acl = disabled
	bootstrap = false
	known_datacenters = 0
	leader = false
	leader_addr = 10.200.2.72:8300
	server = true
raft:
	applied_index = 612523883
	commit_index = 612523883
	fsm_pending = 0
	last_contact = 14.538922ms
	last_log_index = 612523883
	last_log_term = 84769
	last_snapshot_index = 612519767
	last_snapshot_term = 84769
	latest_configuration = [{Suffrage:Voter ID:c07d93ea-8d79-bdd3-4564-fbbd5a35ca97 Address:10.200.3.166:8300} {Suffrage:Voter ID:36660a37-bc45-8acf-d294-e5693bfe19fa Address:10.200.2.72:8300} {Suffrage:Voter ID:673e0ff3-815a-4c56-0b5b-1ccf17d7a7d1 Address:10.200.1.154:8300}]
	latest_configuration_index = 611649367
	num_peers = 2
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Follower
	term = 84769
runtime:
	arch = amd64
	cpu_count = 8
	goroutines = 101349
	max_procs = 4
	os = linux
	version = go1.11.4
serf_lan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 17468
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 1417
	member_time = 5141455
	members = 3299
	query_queue = 0
	query_time = 357
```

</details>

### Operating system and Environment details
Ubuntu 14.04.5 LTS

**Comment by pierresouchay** [comment]
@orarnon  We had similar issues in the past that have been fixed by by a series of patches, with the last one being https://github.com/hashicorp/consul/pull/5520 (included in 1.5.x+ releases if I am not mistaken)
While the issue seems a bit different, it happens this kind of mechanism is registering "wrong" name with same IP while using constantly the "old" name. I wonder if you might have been a victim of it (for instance if you are re-using quickly the same IPs).

Question: do you use node-ids in your config ?

**Comment by orarnon** [comment]
Hi @pierresouchay 
If you mean during Consul registration, then no, we don't use node-ids.

**Comment by ChipV223** [comment]
Hi! @orarnon !

We're closing this issue as there has been no activity in the last 30 days, but if you still having this issue, please re-open the ticket with a comment. As @pierresouchay mentioned previously, this issue was patched in the v1.5.x series and given that you are using v1.4.4, please try upgrading to see if that resolves the issue!
